### PR TITLE
ipv6_addr: remove ng_ prefix

### PR DESCRIPTION
**Pull request by authmillenon**
```sh
mkdir -p sys/include/net/ipv6/
git mv sys/include/net/ng_ipv6/addr.h sys/include/net/ipv6/addr.h
git grep --name-only -i 'ng_ipv6[_/]addr' | \
    xargs sed -i -e 's/ng_\(ipv6[_/]addr\)\(.\+\)\( \+\/.*\)/\1\2   \3/gI' \
        -e 's/\(#define *\)ng_\(ipv6[_/]addr\)\([^ ]\+\)\( \+\)/\1\2\3   \4/gI' \
        -e 's/ng_\(ipv6[_/]addr\)/\1/gI'
```

+ some hand-edits to `net/ipv6/addr.h` and a new file `net/ipv6.h`

### CODE REVIEW

**Comment by authmillenon** [comment]
Biggest change in the hand-edits in particular was to make the flags defines instead of enums.

**Comment by OlegHahm** [comment]
> Biggest change in the hand-edits in particular was to make the flags defines instead of enums.

Why was this necessary?

File: sys/include/net/ng_ipv6/hdr.h [hunk 1]
```diff
@@ -92,8 +92,8 @@ typedef struct __attribute__((packed)) {
     network_uint16_t len;   /**< payload length of this packet. */
     uint8_t nh;             /**< type of next header in this packet. */
     uint8_t hl;             /**< hop limit for this packet. */
-    ng_ipv6_addr_t src;     /**< source address of this packet. */
-    ng_ipv6_addr_t dst;     /**< destination address of this packet. */
+    ipv6_addr_t src;        /**< source address of this packet. */
+    ipv6_addr_t dst;        /**< destination address of this packet. */
 } ng_ipv6_hdr_t;
```
Comment (OlegHahm) [inline sys/include/net/ng_ipv6/hdr.h]: I guess this struct is sensible to non-gnrc implementations as well. Or do you want to separate these things in a separate PR?

**Comment by OlegHahm** [comment]
There are also several defines in `sys/include/net/ng_ipv6/nc.h` that could be re-usable for other implementations of IPv6.

File: sys/include/net/ng_ndp/types.h [hunk 2]
```diff
@@ -209,7 +209,7 @@ typedef struct __attribute__((packed)) {
     network_uint32_t valid_ltime;   /**< valid lifetime */
     network_uint32_t pref_ltime;    /**< preferred lifetime */
     network_uint32_t resv;          /**< reserved field */
-    ng_ipv6_addr_t prefix;          /**< prefix */
+    ipv6_addr_t prefix;             /**< prefix */
```
Comment (OlegHahm) [inline sys/include/net/ng_ndp/types.h]: This file looks also like a good candidate for a non-gnrc header.

**Comment by authmillenon** [comment]
Just to rephrase the title of this PR: this is just for IPv6 addresses. (Extension) headers are in #3594

**Comment by authmillenon** [comment]
>> Biggest change in the hand-edits in particular was to make the flags defines instead of enums.

> Why was this necessary?

To adhere to the coding conventions introduced a while ago.

**Comment by OlegHahm** [comment]
> To adhere to the coding conventions introduced a while ago. 

Ok, but next time I would prefer a separate PR for that.

**Comment by OlegHahm** [comment]
ACK

**Comment by authmillenon** [comment]
Changed and immediately squashed (and it was my mistake ^^").

**Comment by authmillenon** [comment]
Squashed

**Comment by PeterKietzmann** [comment]
Happy Travis!

**Comment by OlegHahm** [comment]
Now a rebase is required. Why did noone hit the merge button if Travis was happy?

**Comment by PeterKietzmann** [comment]
Sorry, PR was too big and I didn't even had a look at the code. Will do so next time.

**Comment by authmillenon** [comment]
Rebased.
